common/dynamicconfig/collection.go

```diff
@@ -64,29 +64,35 @@ type (
 	//   TaskQueue func(namespace string, taskQueue string, taskType enumspb.TaskQueueType)  (matching task queue)
 	//   TaskType func(taskType enumspsb.TaskType)  (history task type)
 	//   ShardID func(shardID int32)
-	BoolPropertyFn                          func() bool
-	BoolPropertyFnWithNamespaceFilter       func(namespace string) bool
-	BoolPropertyFnWithNamespaceIDFilter     func(namespaceID string) bool
-	BoolPropertyFnWithTaskQueueFilter       func(namespace string, taskQueue string, taskType enumspb.TaskQueueType) bool
-	DurationPropertyFn                      func() time.Duration
-	DurationPropertyFnWithNamespaceFilter   func(namespace string) time.Duration
-	DurationPropertyFnWithNamespaceIDFilter func(namespaceID string) time.Duration
-	DurationPropertyFnWithShardIDFilter     func(shardID int32) time.Duration
-	DurationPropertyFnWithTaskQueueFilter   func(namespace string, taskQueue string, taskType enumspb.TaskQueueType) time.Duration
-	DurationPropertyFnWithTaskTypeFilter    func(task enumsspb.TaskType) time.Duration
-	FloatPropertyFn                         func() float64
-	FloatPropertyFnWithNamespaceFilter      func(namespace string) float64
-	FloatPropertyFnWithShardIDFilter        func(shardID int32) float64
-	FloatPropertyFnWithTaskQueueFilter      func(namespace string, taskQueue string, taskType enumspb.TaskQueueType) float64
-	IntPropertyFn                           func() int
-	IntPropertyFnWithNamespaceFilter        func(namespace string) int
-	IntPropertyFnWithShardIDFilter          func(shardID int32) int
-	IntPropertyFnWithTaskQueueFilter        func(namespace string, taskQueue string, taskType enumspb.TaskQueueType) int
-	MapPropertyFn                           func() map[string]any
-	MapPropertyFnWithNamespaceFilter        func(namespace string) map[string]any
-	StringPropertyFn                        func() string
-	StringPropertyFnWithNamespaceFilter     func(namespace string) string
-	StringPropertyFnWithNamespaceIDFilter   func(namespaceID string) string
+	BoolPropertyFn                           func() bool
+	BoolPropertyFnWithNamespaceFilter        func(namespace string) bool
+	BoolPropertyFnWithNamespaceIDFilter      func(namespaceID string) bool
+	BoolPropertyFnWithTaskQueueFilter        func(namespace string, taskQueue string, taskType enumspb.TaskQueueType) bool
+	BoolPropertyFnWithDestionationFilter     func(namespaceID string, destination string) bool
```

Comment:
```suggestion
	BoolPropertyFnWithDestinationFilter     func(namespaceID string, destination string) bool
```

Reply:
Also, you can just add the ones you need for now to keep this shorter. My refactor PR will generate the rest

Reply:
I'd rather just add everything in this PR to avoid having to add these definitions in other PR's that actually create the dynamic config.